packages/loot-design/src/components/modals/ImportTransactions.js

```diff
@@ -43,6 +43,14 @@ export function parseDate(str, order) {
   }
 
   const dateGroups = (a, b) => str => {
+    const parts = str
+      .replace(/^[^\d]+/, '')
+      .replace(/[^\d]+$/, '')
+      .split(/[^\d]+/);
+    if (parts.length >= 3) {
+      return parts.slice(0, 3);
+    }
+
     const digits = str.replace(/[^\d]/g, '');
```

Comment:
Keeping the old logic for backwards compatability. The new logic allows all sorts of weird formats, but it does not allow this one: `20230224` (this is todays date all concatenated in a single string)